MyPy type checking is failing

I see see two errors coming by.  

1. On my machine:

```
> mypy
.venv\Lib\site-packages\sphinx\util\docutils.py:53: error: Type statement is only supported in Python 3.12 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

2. In the latest CI: (https://github.com/sphinx-doc/sphinx-argparse/actions/runs/19854855991/job/56890142738?pr=79#step:7:13)

```
sphinxarg/ext.py:14: error: Module "sphinx.ext.autodoc" has no attribute "mock"  [attr-defined]
```

(Which relates to)
- #82

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -38,9 +38,8 @@ jobs:
     - name: Format with Ruff
       run: ruff format . --diff
 
-#    - name: Type check with MyPy
-#      run: mypy
-# Skip MyPy check for now, see https://github.com/sphinx-doc/sphinx-argparse/issues/85
+    - name: Type check with MyPy
+      run: mypy
 
   twine:
     name: "Check packing with Twine"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -96,7 +96,7 @@ multi_line_output = 3
 
 [tool.mypy]
 files = ["sphinxarg", "test"]
-python_version = "3.10"
+python_version = "3.13"
 exclude = '''(?x)( ^test/roots | ^docs )'''
 
 [[tool.mypy.overrides]]

--- a/sphinxarg/ext.py
+++ b/sphinxarg/ext.py
@@ -5,7 +5,7 @@ import operator
 import os
 import sys
 from argparse import ArgumentParser
-from typing import TYPE_CHECKING, ClassVar, cast
+from typing import TYPE_CHECKING, cast
 
 from docutils import nodes
 from docutils.frontend import get_default_settings
@@ -892,9 +892,7 @@ class ArgParseDomain(Domain):
         'command': XRefRole(),
     }
     indices = []
-    initial_data: ClassVar[
-        dict[str, list[_ObjectDescriptionTuple] | dict[str, list[_ObjectDescriptionTuple]]]
-    ] = {
+    initial_data = {
         'commands': [],
         'commands-by-group': {},
     }

--- a/test/test_default_html.py
+++ b/test/test_default_html.py
@@ -1,6 +1,7 @@
 """Test the HTML builder and check output against XPath."""
 
 import posixpath
+from typing import Any
 
 import pytest
 from sphinx.util.inventory import InventoryFile
@@ -90,8 +91,8 @@ def test_index_is_optional(app, cached_etree_parse):
     assert command_index_file.exists() is False
 
 
-def get_inv_command_uri(inv: dict, field: str) -> str:
-    command = inv.get('commands:command')
+def get_inv_command_uri(inv: dict[str, dict[str, Any]], field: str) -> str:
+    command = inv['commands:command']
     value = command.get(field, None)
     assert value is not None
     if isinstance(value, tuple):

